Spring-06

Max-Pochepets opened the pull request:
Further implementation of Spring MVC

Max-Pochepets committed:
Initial

Max-Pochepets committed:
Refactor:

changed get() in CinemaHallService and MovieService

Max-Pochepets committed:
Refactor:

changed get() in CinemaHallService, MovieService and UserService

Max-Pochepets committed:
checkstyle

Max-Pochepets committed:
Refactor:

get() in CinemaHall and Movie DAO's simplified

Max-Pochepets committed:
Refactor:

get() in MovieSession and User DAO's simplified

Max-Pochepets committed:
removed model from packages being scanned

Max-Pochepets committed:
Fix:

added Id field to MovieSessionRequestDto

Max-Pochepets committed:
Fix:

chery-picking changes from Spring-05

Max-Pochepets committed:
checkstyle

Max-Pochepets committed:
Fix:

changed MovieSessionController logic
chain of changes goes down until the DAO layer

KseniiaMakarova commented on src/main/java/cinema/shop/controller/AuthenticationController.java:
```diff
@@ -0,0 +1,23 @@
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/")
```
we can remove this

KseniiaMakarova commented on src/main/java/cinema/shop/controller/UserController.java:
```diff
@@ -0,0 +1,33 @@
+    public UserResponseDto findByEmail(@RequestParam String email) {
+        Optional<User> optionalUser = userService.findByEmail(email);
+        if (optionalUser.isEmpty()) {
+            return null;
```
let's better throw an exception

KseniiaMakarova commented on src/main/java/cinema/shop/model/dto/response/OrderResponseDto.java:
```diff
@@ -0,0 +1,42 @@
+
+public class OrderResponseDto {
+    private Long id;
+    private List<Long> ticketsId;
```
```suggestion
    private List<Long> ticketIds;
```

KseniiaMakarova commented on src/main/java/cinema/shop/model/dto/response/ShoppingCartResponseDto.java:
```diff
@@ -0,0 +1,33 @@
+
+public class ShoppingCartResponseDto {
+    private Long id;
+    private List<Long> ticketsId;
```
same

KseniiaMakarova commented on src/main/java/cinema/shop/service/impl/MovieSessionServiceImpl.java:
```diff
@@ -25,13 +26,22 @@ public MovieSession add(MovieSession movieSession) {
+    public MovieSession get(Long id) {
+        Optional<MovieSession> optionalMovieSession = movieSessionDao.get(id);
+        if (optionalMovieSession.isEmpty()) {
+            return null;
```
let's throw an exception instead

KseniiaMakarova commented on src/main/java/cinema/shop/service/impl/MovieServiceImpl.java:
```diff
@@ -22,11 +22,8 @@ public Movie add(Movie movie) {
-        }
-        return optionalMovie.get();
+        return movieDao.get(id).orElseThrow(()
+                -> new NoSuchElementException("There is no such movie."));
```
let's have the same approach for all the entities maybe?

KseniiaMakarova commented on src/main/java/cinema/shop/service/dto/mapping/impl/response/UserResponseMapper.java:
```diff
@@ -0,0 +1,19 @@
+
+        responseDto.setId(object.getId());
+        responseDto.setEmail(object.getEmail());
+
```
remove redundant empty lines - the code is harder to read

KseniiaMakarova commented on src/main/java/cinema/shop/service/dto/mapping/impl/response/UserResponseMapper.java:
```diff
@@ -0,0 +1,19 @@
+@Component
+public class UserResponseMapper implements DtoResponseMapper<UserResponseDto, User> {
+    @Override
+    public UserResponseDto toDto(User object) {
```
```suggestion
    public UserResponseDto toDto(User user) {
```

KseniiaMakarova commented on src/main/java/cinema/shop/service/dto/mapping/impl/response/ShoppingCartResponseMapper.java:
```diff
@@ -0,0 +1,25 @@
+public class ShoppingCartResponseMapper
+        implements DtoResponseMapper<ShoppingCartResponseDto, ShoppingCart> {
+    @Override
+    public ShoppingCartResponseDto toDto(ShoppingCart object) {
```
same

KseniiaMakarova commented on src/main/java/cinema/shop/service/dto/mapping/impl/response/ShoppingCartResponseMapper.java:
```diff
@@ -0,0 +1,25 @@
+
+        responseDto.setId(object.getId());
+        responseDto.setUserEmail(object.getUser().getEmail());
+        responseDto.getTicketsId().addAll(object.getTickets().stream()
```
```suggestion
        responseDto.setTicketIds(object.getTickets().stream()
```

KseniiaMakarova commented on src/main/java/cinema/shop/service/dto/mapping/impl/response/OrderResponseMapper.java:
```diff
@@ -0,0 +1,27 @@
+@Component
+public class OrderResponseMapper implements DtoResponseMapper<OrderResponseDto, Order> {
+    @Override
+    public OrderResponseDto toDto(Order object) {
+        OrderResponseDto responseDto = new OrderResponseDto();
+
+        responseDto.setId(object.getId());
+        responseDto.getTicketsId().addAll(object.getTickets().stream()
```
same comments here

KseniiaMakarova commented on src/main/java/cinema/shop/service/dto/mapping/impl/response/OrderResponseMapper.java:
```diff
@@ -0,0 +1,27 @@
+                                                .map(Ticket::getId)
+                                                .collect(Collectors.toList()));
+        responseDto.setOrderDate(object.getOrderDate().format(DateTimeFormatter
+                                                                .ofPattern("dd.MM.yyyy HH:mm")));
```
move the pattern to a constant field

Max-Pochepets committed:
Fix:

removed redundant URL mapping in AuthenticationController
Service methods get() now throw exception instead of returning null
renamed input parameters in ResponseMappers
changed the logic adding tickets id's into ResponseDTO's

Max-Pochepets committed:
Fix:

TicketDaoImpl.add(): persist() -> save()

KseniiaMakarova commented on src/main/java/cinema/shop/controller/AuthenticationController.java:
```diff
@@ -0,0 +1,23 @@
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping
```
what is the purpose of this annotation here?

KseniiaMakarova commented on src/main/java/cinema/shop/service/dto/mapping/impl/response/OrderResponseMapper.java:
```diff
@@ -0,0 +1,28 @@
+
+@Component
+public class OrderResponseMapper implements DtoResponseMapper<OrderResponseDto, Order> {
+
```
redundant empty line

KseniiaMakarova commented on src/main/java/cinema/shop/service/impl/MovieSessionServiceImpl.java:
```diff
@@ -25,13 +26,19 @@ public MovieSession add(MovieSession movieSession) {
+    @Override
+    public MovieSession get(Long id) {
+        return movieSessionDao.get(id).orElseThrow(()
+                -> new NoSuchElementException("Could not get user by id " + id + "."));
```
why user?

KseniiaMakarova reviewed (approved):
see a few comments above

Max-Pochepets committed:
Refactor:

few minor changes
